lib/velum/salt.rb

```diff
@@ -72,11 +72,25 @@ def self.accept_minion(minion_id: "")
 
     # Removes a minion from the cluster
     def self.remove_minion(minion_id: "")
-      res = perform_request(endpoint: "/", method: "post",
-                            data: { client: "wheel",
-                                    fun:    "key.delete",
-                                    match:  minion_id })
-      JSON.parse(res.code)
+      # cloud minions need to be terminated, which will also
+      # remove its salt key from the master
+      res = perform_request(endpoint: "/", method:   "post",
+                            data:     { client: "local",
+                                        tgt:    "admin",
+                                        fun:    "cloud.query" })
+      if res.body.include?(minion_id)
+        res = perform_request(endpoint: "/", method:   "post",
+                              data:     { client: "local_async",
+                                          tgt:    "admin",
+                                          fun:    "cloud.destroy",
+                                          arg:    [ minion_id ] })
```

Comment:
I looked at the salt documentation and it's not clear to me whether this API call is going to drop the minion key as well.

I wonder if we shouldn't always execute the other branch of this `if/else` statement.

Reply:
`cloud.destroy` does remove the minion key.